pkg/asset/agent/installconfig.go

```diff
@@ -82,6 +86,26 @@ func (a *OptionalInstallConfig) validateSupportedPlatforms(installConfig *types.
 	return allErrs
 }
 
+func (a *OptionalInstallConfig) validateSupportedArchs(installConfig *types.InstallConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	fieldPath := field.NewPath("ControlPlane", "Architecture")
+
+	if string(installConfig.ControlPlane.Architecture) != types.ArchitectureAMD64 {
```

Comment:
nit: we will shortly want to expand this to include at least arm64 and ppc64le, so to make that easier this could be a switch statement. (To make it even easier, it could be in a utility function so it only needs to be updated in one place.)

Reply:
It looks like the info from Jira and the review comments are, are quite contradictory. For example, Jira says - 
```
Currently the agent installer supports only x86_64 arch. 
The image creation command must fail if some other arch is configured different from x86_64

We want to have an allowed list of architectures.

allowed = ['x86_64', 'amd64']
```